pandas/core/indexes/datetimes.py

```diff
@@ -1136,6 +1137,11 @@ def union(self, other):
         y : Index or DatetimeIndex
         """
         self._assert_can_do_setop(other)
+
```

Comment:
can you just call the super ``.union()`` here (after any conversion isneeded)? then you can inline the union_corner bizness

Reply:
done